modify rawexec TaskConfig and Config to accept envvar denylist (#25511)
hashicorp/nomad: website/content/docs/drivers/raw_exec.mdx

  must be an absolute path. This will also change the working directory when
  using `nomad alloc exec`.

> - `denied_envvars` - (Optional) Passes a list of environment variables to be
>   scrubbed from the task environment. Supports simple globbing, "*" wildcard
>   accepted as prefix and/or suffix.

Reviewer: ```suggestion
- `denied_envvars` - (Optional) Passes a list of environment variables that
  the driver should scrub from all task environments. Supports globbing "*",
  with wildcard accepted as prefix and/or 
```

style: use active voice; also we don't use words like "simple" so I removed it rather than substitute "basic"

I'm not sure if the driver scrubs the variables or Nomad scrubs the variables

}
```

> - `denied_envvars` - (Optional) Passes a list of environment variables to be
>   scrubbed from all task environments. Supports simple globbing, "*" wildcard
>   accepted as prefix and/or suffix.

Reviewer: ```suggestion
- `denied_envvars` - (Optional) Passes a list of environment variables that
  the driver should scrub from all task environments. Supports globbing "*",
  with wildcard accepted as prefix and/or suffix.
```

style: use active voice; also we don't use words like "simple" so I removed it rather than substitute "basic"

I'm not sure if the driver scrubs the variables or Nomad scrubs the variables

```
config {
  denied_envvars = "["AWS_SECRET_KEY", "*_TOKEN]

> }

Reviewer 2: I can't figure out how to add the triple backtick as a suggestion, but we need to close this code block so that the rest of the page isn't in the example. See https://nomad-git-f-net-12348-rawexecdenylist-hashicorp.vercel.app/nomad/docs/drivers/raw_exec#denied_envvars-1 for the rendered version.

> 
> ```
> config {
>   denied_envvars = "["AWS_SECRET_KEY", "*_TOKEN]
> }

Reviewer: ````suggestion

```hcl
config {
  denied_envvars = "["AWS_SECRET_KEY", "*_TOKEN"]"
}
```

````

@tgross four backticks + suggestion if your suggestion body includes 3 backticks
```
````suggestion
```

Also - I think there are some missing " in the value?? 
```
"["AWS_SECRET_KEY", "*_TOKEN]
```
so I added them - please doublecheck

>   denied_envvars = "["AWS_SECRET_KEY", "*_TOKEN"]

Reviewer 3: looks like there's still a stray `"`

```suggestion
  denied_envvars = ["AWS_SECRET_KEY", "*_TOKEN"]
```
